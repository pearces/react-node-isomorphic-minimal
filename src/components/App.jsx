import React, { useContext } from 'react';
import { Outlet } from 'react-router-dom';
import { ThemeContext } from '../context/ThemeContext';
import './App.scss';

const App = () => {
  const { theme } = useContext(ThemeContext);
  return (
    <main className={theme}>
      <Outlet />
    </main>
  );
};

export default App;
