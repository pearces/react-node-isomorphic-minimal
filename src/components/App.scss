@use '../sass' as *;
@use '~normalize.css/normalize';

#app {
  main {
    padding-left: 0.5rem;
    padding-right: 0.5rem;
    height: 100vh;

    &.dark {
      background-color: #222;
      color: #fff;
    }
  }
}

button {
  &:not(disabled) {
    cursor: pointer;
  }

  display: inline-block;
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  user-select: none;
  border: 1px solid transparent;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition:
    color 0.15s ease-in-out,
    background-color 0.15s ease-in-out,
    border-color 0.15s ease-in-out,
    box-shadow 0.15s ease-in-out;
  width: 100%;

  max-width: 90%;
  @include sm {
    max-width: 80%;
  }
  @include md {
    max-width: 50%;
  }
  @include lg {
    max-width: 33%;
  }
  @include xl {
    max-width: 25%;
  }

  &.primary {
    color: #fff;
    background-color: #337ab7;
    border-color: #2e6da4;

    &:hover {
      background-color: #286090;
      border-color: #204d74;
    }
  }

  &.center {
    margin: 0 auto;
    display: block;
  }
}
