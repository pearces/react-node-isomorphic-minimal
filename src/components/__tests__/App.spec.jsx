import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

describe('NoMatch', () => {
  it('matches component snapshot', () => {
    const html = renderer.create(
      <App>
        <div />
      </App>
    );
    const tree = html.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
