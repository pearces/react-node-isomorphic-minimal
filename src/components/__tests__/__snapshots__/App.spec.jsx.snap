// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch matches component snapshot 1`] = `
<main
  className="light"
/>
`;
